### PR TITLE
[ABSL] Use absl::Substitute in rigid_transform debug messages.

### PR DESCRIPTION
**Pull request by pifon2a**
it is at least readable.

### CODE REVIEW

File: cartographer/transform/rigid_transform.h [hunk 1]
```diff
@@ -169,23 +163,10 @@ class Rigid3 {
...
-    out.append(std::to_string(rotation().w()));
-    out.append(", ");
-    out.append(std::to_string(rotation().x()));
-    out.append(", ");
-    out.append(std::to_string(rotation().y()));
-    out.append(", ");
-    out.append(std::to_string(rotation().z()));
-    out.append("] }");
-    return out;
+    return absl::Substitute("{ t: [$0, $1, $2], r: [$3, $4, $5, $6] }",
```
Comment (gaschler) [inline cartographer/transform/rigid_transform.h]: don't change "q:" to "r:"

Comment (pifon2a) [reply cartographer/transform/rigid_transform.h]: oh, thanks for noticing

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Pull request merged**